site-scrapers/CVS.js

```diff
@@ -0,0 +1,47 @@
+const sites = require("../data/sites.json");
+const https = require("https");
+const crypto = require("crypto");
+const { map } = require(".");
+
+const siteName = "CVS";
+const site = sites[siteName];
+
+module.exports = async function GetAvailableAppointments(browser) {
+    console.log(`${siteName} starting.`);
+    const webData = await ScrapeWebsiteData(browser);
+    console.log(`${siteName} done.`);
+    return webData.map((responseLocation) => {
+        let hasAvailability = parseInt(responseLocation.totalAvailable)
+            ? true
+            : false;
+        let availability = responseLocation.totalAvailable;
+        return {
+            name: `${siteName} (${responseLocation.city})`,
+            hasAvailability,
+            availability,
+            signUpLink: site.website,
+            ...responseLocation,
+        };
+    });
+};
+
+async function ScrapeWebsiteData(browser) {
+    let rawData = {};
+    const page = await browser.newPage();
+    await page.goto(site.website, { waitUntil: "domcontentloaded" });
+    await page.waitForSelector("a[data-modal='vaccineinfo-MA']");
+
+    const [searchResponse, ...rest] = await Promise.all([
+        Promise.race([
+            page.waitForResponse(
+                "https://www.cvs.com/immunizations/covid-19-vaccine.vaccine-status.ma.json?vaccineinfo"
+            ),
+            page.waitForNavigation(),
+        ]),
+        page.click("a[data-modal='vaccineinfo-MA']"),
+    ]);
+    const response = (await searchResponse.buffer()).toString();
+    responseJSON = JSON.parse(response);
+    rawData = responseJSON.responsePayloadData.data.MA;
```

Comment:
We could supplement with actual locations, but I guess that's a 2.0 feature.

Reply:
Right. When there are spots available I'll see how difficult it is to get more granular data.